analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -130,142 +116,159 @@ Because `DBI` does not support full_join, we had to use a series of `left_join`
+# Add up how many callers identify each mutation, but ignore VarDict
+consensus_count <- rowSums(detect_mat[, colnames(detect_mat) != "VAF_vardict"])
+
+# Remove the 0's
+consensus_count <- consensus_count[which(consensus_count != 0)]
 ```
 
-## Upset graph
-
-Make the upset plot. 
+Print out the numbers for the three callers' consensus.
 
 ```{r}
-upsettr_plot <- callers_per_mutation %>%
-  upset_plot()
-
-# Print this out here
-upsettr_plot
+cat(" At least 1 out of 3:", length(consensus_count), "\n",
+    "At least 2 out of 3:", sum(consensus_count >= 2), "\n",
+    "3 out of 3:", sum(consensus_count == 3))
 ```
 
-Save this plot to a png
+## Upset graphs
+
+Make the upset plot. 
 
 ```{r}
-# We can save the plot like a normal ggplot
-ggplot2::ggsave(plot = upsettr_plot, 
-                file.path(plots_dir, "pbta-upset_plot.png"))
+upset_png(detect_mat, 
+          plot_file_path = file.path(plots_dir, "pbta-upset-plot.png"))
 ```
 
+![Upset plot - all data](plots/pbta-caller-comparison/pbta-upset-plot.png)
+
 Transcripts only upset plot. 
 
 ```{r}
-callers_per_mutation %>% 
-  dplyr::filter(!(Variant_Classification %in% c("5'Flank", "3'Flank", "IGR", "Intron"))) %>%
-  upset_plot()
+transcripts <- !(all_caller_df$Variant_Classification %in% c("5'Flank", "3'Flank", "IGR", "Intron"))
+
+upset_png(detect_mat, 
+          plot_file_path = file.path(plots_dir, "pbta-upset-plot-transcipts-only.png"), 
+          subset_vector = transcripts)
 ```
 
+![Upset plot - transcripts only](plots/pbta-caller-comparison/pbta-upset-plot-transcipts-only.png)
+
 Non transcript upset plot. 
 
 ```{r}
-callers_per_mutation %>% 
-  dplyr::filter(Variant_Classification %in% c("5'Flank", "3'Flank", "IGR", "Intron")) %>%
-  upset_plot()
+not_transcripts <- all_caller_df$Variant_Classification %in% c("5'Flank", "3'Flank", "IGR", "Intron") 
+
+upset_png(detect_mat,
+          plot_file_path = file.path(plots_dir, "pbta-upset-plot-not-transcipts-only.png"), 
+          subset_vector = transcripts)
 ```
 
-## VAF for combination sets of callers
+![Upset plot - transcripts only](plots/pbta-caller-comparison/pbta-upset-plot-not-transcipts-only.png)
+
+## VAF distribution for each caller
 
-Graph mutations by combinations of callers and plot the VAF density.
-Get list of callers per mutation and the median vaf for each. 
-Graph the median VAF for each combination of callers.
+Create consensus VAF based on strelka's VAFs and which mutations would be included
+
+```{r}
+# Make new VAF column based on strelka's VAFs
+all_caller_df$VAF_consensus <- all_caller_df$VAF_strelka
+
+# Figure out which mutations are not in the consensus
+not_consensus <- dimnames(consensus_count)[[1]][which(consensus_count < 3)]
+
+# Make non consensus mutations in consensus)VAF an NA
+all_caller_df$VAF_consensus[not_consensus] <- NA
+```
+
+Make a long form VAF data.frame for exploring mutations VAFs from each caller. 
+
+```{r}
+vaf_df <- all_caller_df %>% 
```

Comment:
this is still going to include vardict, correct?

Reply:
Yes.